packages/server/src/cloud/gcp/storage.ts

```diff
@@ -15,18 +15,37 @@ export class GoogleCloudStorage extends BaseBinaryStorage {
     this.bucket = this.storage.bucket(bucketName);
   }
 
-  async writeFile(key: string, contentType: string | undefined, stream: BinarySource): Promise<void> {
+  /**
+   * Writes a file to Google Cloud Storage.
+   * This method now correctly handles both Readable streams and strings as input.
+   * @param key - The key (path) for the file in the bucket.
+   * @param contentType - The MIME type of the content.
+   * @param data - The content to write, either as a Readable stream or a string.
+   */
+  async writeFile(key: string, contentType: string | undefined, data: BinarySource): Promise<void> {
     const file = this.bucket.file(key);
-    return new Promise((resolve, reject) => {
-      const writeStream = file.createWriteStream({
-        metadata: { contentType: contentType ?? 'application/octet-stream' },
-        resumable: false,
+    const options = {
+      metadata: { contentType: contentType ?? 'application/octet-stream' },
+      // Resumable uploads are generally recommended for robustness, especially for streams.
+      // You can set this to false if you are certain you will only upload small files.
+      resumable: true,
+    };
+
+    if (isString(data)) {
+      // For strings, the file.save() method is the most direct and simplest way.
+      // It handles encoding and buffering automatically.
+      await file.save(data, options);
+    } else {
+      // For streams, we pipe the data to a write stream provided by the client library.
+      // We wrap this callback-based stream logic in a Promise to work seamlessly with async/await.
+      await new Promise<void>((resolve, reject) => {
+        const writeStream = file.createWriteStream(options);
+        (data as Readable)
```

Comment:
Should we handle the case where `data` is a `Uint8Array` before the else?

Reply:
Oh I think this is a workaround the strict typing of the interface, yeah?

Reply:
Ahh, my bad.  This cast is unnecessary.

> Should we handle the case where data is a Uint8Array before the else?

No, that shouldn't be necessary here, because we are in `server`:

```
export type BinarySource = Readable | string;
```

Which is confusing, because it's different in `core`:

```
export type BinarySource = string | File | Blob | Uint8Array;
```

🙃